ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/filtered_tenants.py

```diff
@@ -237,12 +242,18 @@ def filtered_vrfs(self: SharedUtils, tenant: dict) -> list[dict]:
                 )
             ]
 
+            if self.wan_role is not None and (wan_vni := get(original_vrf, "wan_vni")):
+                # TODO need to make sure the VRF exists on uplink switch otherwise configured
+                # everywhere
+                vrf["wan_vni"] = wan_vni
+
             if (
                 vrf["svis"]
                 or vrf["l3_interfaces"]
                 or vrf["loopbacks"]
                 or "all" in always_include_vrfs_in_tenants
                 or tenant["name"] in always_include_vrfs_in_tenants
+                or (self.wan_role is not None and vrf.get("wan_vni", None) is not None)
```

Comment:
is_interesting_uplink_vrf(vrf_name) -> bool
"""
attract VRF defined on uplink switch - most likely needed to extend over the WAN
"""

Reply:
need another function `is_allowed_vrf` 

filter for VRFs to prune any VRF (at the top of the for loop) - list of VRF names / takes precedence over always include vrfs in tenants